chidioguejiofor/airtech-api
# #165062683: Admin create flight

#### What does this PR achieve?
- creates Django apps
- adds Flight model and migration to the app
- updates User migrations
- updates clean_up_user_data function to tokenise only id, username and email
- adds the implementation of POST /flight
- adds Token Validation to the app
- adds tests for the POST /flight
- ensures that all added tests are passing


#### Any background context?
- None

#### New environmental Variables?
- None

#### Manual Testing Steps
- Migrate your data to your database via `python manage.py makemigrations && python manage.py migrate`
- Signup via /api/v1/signup to add a user to the app
- Manually  change the attribute of the user to `admin`
- Hit the POST '/api/v1/flight' endpoint with the following request header and body:

*Header*
```
{ 'Authorization`: `Bearer <token>'}
```
*Body*
```
{
      "capacity": <capacity>,
      "location": <location>,
      "destination": <destination>,
      "schedule": <schedule>,
      "amount": <amount>,
      "type": <type>
}
```

You should get  this response


```
{
    "status": "success",
    "message": "Flight created successfully",
    "data": {
        "id": "bc5f1c22-4b10-4255-aa6e-1e66678e1c48",
        "capacity": 70,
        "location": "Popo York",
        "destination": "Lagos, Nigeria",
        "schedule": "2017-10-10T14:00:01Z",
        "amount": "4000.70",
        "type": "international",
        "creator": {
            "id": "8b963656-295d-48b0-92c6-6d0876ad1904",
            "firstName": "Prof",
            "gender": "Male",
            "lastName": "fdajhfio",
            "email": "prof1@email11.com",
            "username": "Prof10111"
        },
        "createdAt": "2019-04-06T23:17:18.658113Z",
        "updatedAt": null
    }
}
```
